instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -1,13 +1,54 @@
 import React from 'react';
 import './CommentSection.css';
+import moment from 'moment';
 
 
 class CommentSection extends React.Component {
+   
+    constructor(props) {
+        super(props)
+        this.addNewComment = this.addNewComment.bind(this);
+        this.state = {
+            comments : this.props.comments.map(comment => comment.text),
+            username : this.props.comments.map(comment => comment.username)
+        }
+    }
 
-    render() {
+    addNewComment(e) {
+        e.preventDefault();
+        // console.log(e, 'this comment has been entered');
+        // console.dir(e.target.children[0].value);
+        // console.dir(this.state.comments);
+        this.setState( {
+            comments : this.state.comments.concat(e.target.children[0].value),
+            username : this.state.username.concat('newuser')
+        })
+
+    }
 
+    render() {
+        const date = moment('2019-04-01').fromNow()
         return (
-            <input id='comment' placeholder='Add a comment'/>
+            <div id='commentsection'>
+                
+                <div className='users divvy'>
```

Comment:
Interesting approach to render the usernames separate from the comment. I like it!